### PR TITLE
Record more status codes

### PR DESCRIPTION
**Pull request by schwers**
We've seen an increase in 404's, and bug reports of 'not found' pages. 404's are now down in grafana after https://github.com/reddit/reddit-mobile/pull/1030 was merged. But there are continued reports of bad pages. 

Noteable fixes:
* listing and comments pages should now always dispatch `success` or `failed`, this should ensure we always update the status code properly.
* errorLogger middleware is updated to ensure we properly log comments page failures
* ads requests  aren't made on the server, and as failsafe the adblock check shortcircuits on the server. (this came up in testing SEO renders)
* client's sentryOnError is defaulted to a no-op for dev mode

👓 @curioussavage @prashtx @dwick

### CODE REVIEW

File: src/app/actions/commentsPage.js [hunk 1]
```diff
@@ -69,11 +69,7 @@ export const fetchCommentsPage = commentsPageParams => async (dispatch, getState
     const apiResponse = await CommentsEndpoint.get(apiOptions, commentsPageParams);
     dispatch(received(commentsPageId, apiResponse));
   } catch (e) {
-    if (e instanceof ResponseError) {
-      dispatch(failed(commentsPageId, e));
-    } else {
-      throw e;
-    }
+    dispatch(failed(commentsPageId, e));
```
Comment (curioussavage) [inline src/app/actions/commentsPage.js]: You decided there was no need to throw again?

File: src/app/reduxMiddleware/errorLogger.js [hunk 2]
```diff
@@ -118,6 +118,10 @@ const checkForSpecificErrors = action => {
       if (error) {
         return error;
       }
+
+      if (action.payload && action.payload.error) {
```
Comment (curioussavage) [inline src/app/reduxMiddleware/errorLogger.js]: I see. This is why you don't need to throw. Nice.

Comment (schwers) [inline src/app/reduxMiddleware/errorLogger.js]: Yup, exactly :) only dispatching simplifies things and ensures our status codes are proper

**Comment by curioussavage** [comment]
🐟  can it go out today?

**Comment by prashtx** [comment]
:fish:

**Pull request merged**